File: app/src/test/java/com/star_zero/example/androidmvvm/helper/HotTestSubscriber.java
package com.star_zero.example.androidmvvm.helper;

import rx.observers.TestSubscriber;


public class HotTestSubscriber<T> extends TestSubscriber<T> {

    private int count;

    public HotTestSubscriber(int count) {
        this.count = count;
    }

    @Override
    public void onNext(T t) {
        super.onNext(t);

        count--;

        // when call onNext specific number of times, subscriber terminate.
        if (count == 0) {
            onCompleted();
        }
    }

    public static <T> HotTestSubscriber<T> create() {
        return new HotTestSubscriber<>(1);
    }

    public static <T> HotTestSubscriber<T> create(int count) {
        return new HotTestSubscriber<>(count);
    }
}

File: app/src/test/kotlin/com/star_zero/example/androidmvvm/helper/HotTestSubscriber.kt
package com.star_zero.example.androidmvvm.helper

import rx.observers.TestSubscriber


class HotTestSubscriber<T>(private var count: Int) : TestSubscriber<T>() {

    override fun onNext(t: T) {
        super.onNext(t)

        count--

        // when call onNext specific number of times, subscriber terminate.
        if (count == 0) {
            onCompleted()
        }
    }

    companion object {

        fun <T> create(): HotTestSubscriber<T> {
            return HotTestSubscriber(1)
        }

        fun <T> create(count: Int): HotTestSubscriber<T> {
            return HotTestSubscriber(count)
        }
    }
}
